### ISSUE TITLE
Build fails in NixOS

### ISSUE THREAD

**Issue by gsychev** [emqx/quic#78]
### Enviroment 

Version: `e850f4feaa907b1e6ea653ff20bde46f4ca33815` (latest commit)

Dockerfile with reproduction steps:
```Dockerfile
FROM nixos/nix
COPY . .
RUN nix-shell -p gcc gnumake cmake rebar3 erlangR23 perl openssl --run make
```

### Actual behavior

Build fails with
```
/nix/store/5ddb4j8z84p6sjphr0kh6cbq5jd12ncs-binutils-2.35.1/bin/ld: cannot find -lATOMIC-NOTFOUND
collect2: error: ld returned 1 exit status
make[3]: *** [msquic/src/bin/CMakeFiles/msquic.dir/build.make:107: msquic/bin/Release/libmsquic.so] Error 1
```

### Desired behavior

Build passes

### Overview

This is caused by buggy `CMakeLists.txt` in `msquic`. This bug was fixed in latest versions of `msquic`

Ref:
https://github.com/microsoft/msquic/blob/v1.8.0/src/inc/CMakeLists.txt#L34
https://github.com/microsoft/msquic/blob/main/src/inc/CMakeLists.txt#L40

**Comment by datafoo** [comment]
I am also having the problem when trying to use `emqx/emqtt` version `1.4.7` in a Elixir project under NixOS:

```sh
[nix-shell:~/dev/myproject]$ mix compile
-- CMAKE Version: 3.21.2
-- Source Dir: /home/me/dev/myproject/deps/quicer/msquic
-- Host System name: Linux
-- Setting policy 0091
-- System name: Linux
-- System version: 5.10.89
-- Platform version: 
-- Build type: RelWithDebInfo
-- QUIC Platform: linux
-- Version Build ID: 0
-- Version Suffix: -private
-- Enabling OpenSsl configuration tests
-- Enabling shared ephemeral port work around
-- Disabling tracing
-- Setting openssldir to /nix/store/ligwjs9jslf1na0mxzycjd1lvpqrr2fn-openssl-1.1.1l/etc/ssl
-- Configuring for OpenSSL
-- Configuring done
-- Generating done
-- Build files have been written to: /home/me/dev/myproject/deps/quicer/c_build
make: Entering directory '/home/me/dev/myproject/deps/quicer/c_build'
make[1]: Entering directory '/home/me/dev/myproject/deps/quicer/c_build'
make[2]: Entering directory '/home/me/dev/myproject/deps/quicer/c_build'
make[2]: Entering directory '/home/me/dev/myproject/deps/quicer/c_build'
make[2]: Leaving directory '/home/me/dev/myproject/deps/quicer/c_build'
[  2%] Built target OpenSSL_Target
make[2]: Entering directory '/home/me/dev/myproject/deps/quicer/c_build'
Consolidate compiler generated dependencies of target core
make[2]: Leaving directory '/home/me/dev/myproject/deps/quicer/c_build'
Consolidate compiler generated dependencies of target platform
make[2]: Leaving directory '/home/me/dev/myproject/deps/quicer/c_build'
[ 53%] Built target core
[ 72%] Built target platform
make[2]: Entering directory '/home/me/dev/myproject/deps/quicer/c_build'
Consolidate compiler generated dependencies of target msquic
make[2]: Leaving directory '/home/me/dev/myproject/deps/quicer/c_build'
make[2]: Entering directory '/home/me/dev/myproject/deps/quicer/c_build'
[ 73%] Linking C shared library ../../bin/Release/libmsquic.so
/nix/store/h19zwlkrp6b0hp3ypbqdcggnyarn3af6-binutils-2.35.2/bin/ld: cannot find -lATOMIC-NOTFOUND
collect2: error: ld returned 1 exit status
make[2]: *** [msquic/src/bin/CMakeFiles/msquic.dir/build.make:101: msquic/bin/Release/libmsquic.so] Error 1
make[2]: Leaving directory '/home/me/dev/myproject/deps/quicer/c_build'
make[1]: *** [CMakeFiles/Makefile2:345: msquic/src/bin/CMakeFiles/msquic.dir/all] Error 2
make[1]: Leaving directory '/home/me/dev/myproject/deps/quicer/c_build'
make: *** [Makefile:136: all] Error 2
make: Leaving directory '/home/me/dev/myproject/deps/quicer/c_build'
===> Hook for compile failed!

** (Mix) Could not compile dependency :quicer, "/home/me/.mix/rebar3 bare compile --paths /home/me/dev/myproject/_build/dev/lib/*/ebin" command failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile quicer", update it with "mix deps.update quicer" or clean it with "mix deps.clean quicer"
```

**Comment by qzhuyan** [comment]
@datafoo as you can see 
```
/nix/store/h19zwlkrp6b0hp3ypbqdcggnyarn3af6-binutils-2.35.2/bin/ld: cannot find -lATOMIC-NOTFOUND
```
The libatomic is missing in the OS

**Comment by gsychev** [comment]
@datafoo , you can just remove `${ATOMIC}` from CMakeLists.txt. I can provide more details, if you want

**Comment by qzhuyan** [comment]
I think it should be solved with latest 0.0.12 release with MsQuic 2.0.2

**Comment by wavewizard** [comment]
use nix package  libgccjit to access libatomic to build quicer.

`nix-shell -p libgccjit rebar3 cmake glibc`
